pkg/planner/core/find_best_task.go

```diff
@@ -2589,6 +2593,23 @@ func convertToTableScan(ds *logicalop.DataSource, prop *property.PhysicalPropert
 		// TiFlash fast mode(https://github.com/pingcap/tidb/pull/35851) does not keep order in TableScan
 		return base.InvalidTask, nil
 	}
+	// When the the store type is TiFlash, we will try to set the UsedColumnarIndexes.
+	// FIXME: Now it is a naive implementation, we will refine it based on selectivity and cost in the future.
+	if ts.StoreType == kv.TiFlash {
+		for _, index := range ts.Table.Indices {
+			if index.State == model.StatePublic && index.InvertedInfo != nil {
+				ts.UsedColumnarIndexes = append(ts.UsedColumnarIndexes, &tipb.ColumnarIndexInfo{
+					IndexType: tipb.ColumnarIndexType_TypeInverted,
+					Index: &tipb.ColumnarIndexInfo_InvertedQueryInfo{
+						InvertedQueryInfo: &tipb.InvertedQueryInfo{
+							IndexId:  index.ID,
+							ColumnId: index.InvertedInfo.ColumnID,
+						},
+					},
+				})
+			}
```

Comment:
do we need to check whether the index is used or not?
if the numeric column a has inverted index while this SQL don't have predicates on it, do we need to add that index here?

Reply:
To simplify, just add all inverted index info now since we all implement a better algorithm soon.